src/PPN-microbench/core_to_core_latency.cpp

```diff
@@ -0,0 +1,110 @@
+#include <PPN-microbench/core_to_core_latency.hpp>
+
+using std::chrono::duration_cast;
+using std::chrono::nanoseconds;
+using std::chrono::steady_clock;
+
+constexpr int preheat = 100;
+
+CoreToCoreLatency::CoreToCoreLatency(int nbMeasures) : Microbench("Core To Core Latency", 900) {
+    this->nbMeasures = nbMeasures;
+    nbCores = context.getCpus();
+    resultsMin.reserve(nbCores * nbCores);
+    resultsMean.reserve(nbCores * nbCores);
+}
+
+CoreToCoreLatency::~CoreToCoreLatency() {}
+
+void CoreToCoreLatency::pinThread(int core) {
+    // To call a thread
+    Context context = Context::getInstance();
+    std::vector<size_t> threadMapping = context.getThreadMapping();
+    cpu_set_t cpusets;
+    CPU_ZERO(&cpusets);
+    CPU_SET(threadMapping[core], &cpusets);
+    if (sched_setaffinity(0, sizeof(cpusets), &cpusets) == -1) {
+        perror("sched_setaffinity");
+        exit(1);
+    }
+}
+
+void CoreToCoreLatency::run() {
+    // Set all threads to 0
+    cpu_set_t cpusets;
+    CPU_ZERO(&cpusets);
+    if (sched_getaffinity(0, sizeof(cpusets), &cpusets) == -1) {
+        perror("sched_getaffinity");
+        exit(1);
+    }
+
+    // Enumerate available cores
+    std::vector<int> cpus;
```

Comment:
Given that you know the number of CPU cores, you can use the pre-allocating ctor for `vector` here.